drivers/usbmisc/fusb302.c

```diff
@@ -0,0 +1,1941 @@
+  if (enable)
+    {
+      regval |= pullup;
+    }
+  else
+    {
+      regval &= ~pullup;
+    }
+
+  fusb302_putreg(priv, FUSB302_SWITCHES0_REG, regval);
+}
+
+/****************************************************************************
+ * Name: enableccmeas
+ *
+ * Description:
+ *   Enable/disable CC measurement
+ *
+ * Input Parameters:
+ *  priv - pointer to device structure
+ *  measure - CC1 or CC2
+ *  enable - enable or disable it
+ *
+ * Returned Value:
+ *   none
+ *
+ ****************************************************************************/
+
+void enableccmeas(struct fusb302_dev_s *priv, enum cc_meas_e measure,
+                  bool enable)
+{
+  int regval;
+
+  regval = fusb302_getreg(priv, FUSB302_SWITCHES0_REG);
+  if (enable)
+    {
+      regval |= measure;
+    }
+  else
+    {
+      regval &= ~measure;
+    }
+
+  fusb302_putreg(priv, FUSB302_SWITCHES0_REG, regval);
+}
+
+/****************************************************************************
+ * Name: setmdac
+ *
+ * Description:
+ *   Set MDAC measurement threshold
+ *
+ * Input Parameters:
+ *  priv      - pointer to device structure
+ *  threhold  - MDAC threshold value
+ *
+ * Returned Value:
+ *   none
+ *
+ ****************************************************************************/
+
+void setmdac(struct fusb302_dev_s *priv, enum src_current_e thresh)
+{
+  int regval;
+
+  regval = fusb302_getreg(priv, FUSB302_MEASURE_REG);
+  regval &= ~MEASURE_MDAC_MASK;
+  regval |= MEASURE_MDAC(src_mdac_val[thresh]);
+
+  fusb302_putreg(priv, FUSB302_MEASURE_REG, regval);
+  usleep(150);
+}
+
+/****************************************************************************
+ * Name: enablevconn
+ *
+ * Description:
+ *   Enable/disable Vconn supply for host mode
+ *
+ * Input Parameters:
+ *  priv - pointer to device structure
+ *  vconn - CC1 or CC2
+ *  enable - enable or disable it
+ *
+ * Returned Value:
+ *   none
+ *
+ ****************************************************************************/
+
+void enablevconn(FAR struct fusb302_dev_s *priv, enum cc_vconn_e vconn,
+                 bool enable)
+{
+  int regval;
+
+  regval = fusb302_getreg(priv, FUSB302_SWITCHES0_REG);
+  if (enable)
+    {
+      regval |= vconn;
+    }
+  else
+    {
+      regval &= ~vconn;
+    }
+
+  fusb302_putreg(priv, FUSB302_SWITCHES0_REG, regval);
+}
+
+/****************************************************************************
+ * Name: set_switches
+ *
+ * Description:
+ *   Set up switches for required operational mode
+ *
+ * Input Parameters:
+ *  priv     - pointer to device structure
+ *  toggsval - the value read after toggling is done
+ *  enable - enable or disable it
+ *
+ * Returned Value:
+ *   none
+ *
+ ****************************************************************************/
+
+void set_switches(FAR struct fusb302_dev_s *priv,
+                                        uint8_t toggsval)
+{
+  switch (toggsval)
+
+  /* what device detect type was seen */
```

Comment:
Suggestion: put the comment befoer "switch (toggsval)" to avoid these lines between switch and case

Reply:
OK